### PR TITLE
Implemented argument to provide import values

### LINKED ISSUE #175: ARN validation failure on imported values
I'm getting validation errors when trying to use `!ImportValue` for properties that expect an ARN.

Sample template:
```
Resources:
  DomainName:
    Type: AWS::ApiGateway::DomainName
    Properties:
      RegionalCertificateArn: !ImportValue ExportedRegionalCertificateArn
      DomainName: some.domain.com
      EndpointConfiguration:
        Types:
          - REGIONAL
```

I get the following result:
```
1 crit
Resource: Resources > DomainName > Properties > RegionalCertificateArn
Message: Expecting an ARN, got 'IMPORTEDVALUEExportedRegionalCertificateArn'
Documentation: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-domainname.html
```

I tried doing something similar with a different type of resource that also expects an ARN:

```
Resources:
  User:
    Type: "AWS::IAM::User"
    Properties: 
      ManagedPolicyArns:
        - !ImportValue ExportedManagedPolicyArn
      UserName: "someuser"
```

This passed, but it passes even if I put in a garbage non-ARN, so it doesn't appear to be doing the validation at all.

I am using `cfn-lint@1.7.2`.

Similar to #143, but so far I don't see a workaround.  I'm happy to work on a PR, if this is indeed erroneous behavior.

Thanks!

### PR DESCRIPTION
**Pull request by jonscheiding**
Closes #175 .

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -59,6 +59,7 @@ program
     .option('--guess-parameters', 'Guess any parameters that are not explicitely passed in and have no Default. This is the default behaviour.')
     .option('-G, --no-guess-parameters', 'Fail validation if a parameter with no Default is not passed')
     .option('-g, --only-guess-parameters <items>', 'Guess the provided parameters, and fail validation if a parameter with no Default is passed', list)
+    .option('-i, --importValues <items>', 'List of import values', list)
```
Comment (martysweet) [inline src/index.ts]: `--import-values`

File: src/test/indexTest.ts [hunk 2]
```diff
@@ -58,6 +59,13 @@ describe('index', () => {
             });
         }).timeout(5000);
 
+        it('validate importValue flag', (done) => {
+            exec('node lib/index.js validate testData/valid/yaml/valid_import_value_type.yaml --importValues ImportedValueOutputPort="20"', function(error, stdout, stderr) {
```
Comment (martysweet) [inline src/test/indexTest.ts]: What if the value is not quoted? Does this handle plain integers well?

File: src/test/indexTest.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ const expect = chai.expect;
 const assert = chai.assert;
 
 import childProcess = require('child_process');
+import { doesNotThrow } from 'assert';
```
Comment (martysweet) [inline src/test/indexTest.ts]: Is this required?

File: src/validator.ts [hunk 4]
```diff
@@ -1162,7 +1173,11 @@ function doIntrinsicImportValue(ref: any, key: string){
 
     // Resolve
     if(typeof toGet == 'string'){
-        return "IMPORTEDVALUE" + toGet; // TODO: Consider making this commandline defined
+        const importValue = importRuntimeOverride[toGet];
+        if(importValue !== undefined) {
+            return importValue;
+        }
+        return "IMPORTEDVALUE" + toGet;
```
Comment (martysweet) [inline src/validator.ts]: Can you add a comment like:
`\\ Backward compatibility, do not raise an error if the import value is not provided.`

File: testData/valid/yaml/valid_import_value_type.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Resources:
+  WebServerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: "Enable HTTP access via port 80 locked down to the load balancer + SSH access"
+      SecurityGroupIngress:
+      - CidrIp: 0.0.0.0/0
+        FromPort:
+          Fn::ImportValue: "ImportedValueOutputPort"
```
Comment (martysweet) [inline testData/valid/yaml/valid_import_value_type.yaml]: Maybe a test for a missing ImportValue could be useful

File: src/test/indexTest.ts (hunk 3 above)
Comment (jonscheiding) [reply src/test/indexTest.ts]: @martysweet Not sure how that got in there.  I'll remove it.

File: src/test/indexTest.ts (hunk 2 above)
Comment (jonscheiding) [reply src/test/indexTest.ts]: I aped the importValues handling code from the code for parameters, so it should work the same.  I added a test that confirms.

**Comment by jonscheiding** [comment]
@martysweet Thanks for the feedback!  I addressed most of your comments, just had a question about one of them.

File: testData/valid/yaml/valid_import_value_type.yaml (hunk 5 above)
Comment (martysweet) [reply testData/valid/yaml/valid_import_value_type.yaml]: Yep looks good

**Comment by martysweet** [comment]
Thanks for this @jonscheiding, will be released shortly as v1.8.0

**Pull request merged**